api/http/handler/stacks/create_swarm_stack.go

```diff
@@ -156,17 +157,19 @@ func (handler *Handler) createSwarmStackFromGitRepository(w http.ResponseWriter,
 
 	gitCloneParams := &cloneRepositoryParameters{
 		url:            payload.RepositoryURL,
+		referenceName:  payload.RepositoryReferenceName,
 		path:           projectPath,
 		authentication: payload.RepositoryAuthentication,
 		username:       payload.RepositoryUsername,
 		password:       payload.RepositoryPassword,
 	}
+	doCleanUp := true
```

Comment:
Actually, the following statement should be located before the call to `handler.cloneGitRepository` like you did in the `create_compose_stack.go`:

```go
doCleanUp := true
defer handler.cleanUp(stack, &doCleanUp)
```

Remove any other `defer handle.cleanUp` calls from this file.

Reply:
Done